app/delivery/send_to_providers.py

```diff
@@ -52,14 +49,14 @@ def send_sms_to_provider(notification):
             show_prefix=service.prefix_sms,
         )
 
+        created_at = notification.created_at
```

Comment:
Just to check, what benefit do you see us getting from not doing the extra select query after the insert?

- It will mean we do one fewer query which is positive, although I don't think of massive impact
- I don't think it will release a DB connection early? Leo mentioned something about lack of DB connection pooling and it being a one to one with celery workers anyway? Or did I misunderstand this?

Reply:
If you look at the calls the database makes you will get the extra query, because if you access a data model object after a commit the object is considered "dirty", so it will re-query to get the "current" data model object. The extra query happens regardless of what's happening with the connection pooling, regardless of the connection being released to the pool or not.